api/base/middleware.py

```diff
@@ -107,11 +109,30 @@ class CorsMiddleware(corsheaders.middleware.CorsMiddleware):
     """
     Augment CORS origin white list with the Institution model's domains.
     """
-    def origin_not_found_in_white_lists(self, origin, url):
-        not_found = super(CorsMiddleware, self).origin_not_found_in_white_lists(origin, url)
-        if not_found:
-            not_found = Institution.find(Q('domains', 'eq', url.netloc.lower())).count() == 0
-        return not_found
+    def process_request(self, request):
+        def origin_not_found_in_white_lists(self, request, origin, url):
+            not_found = super(CorsMiddleware, self).origin_not_found_in_white_lists(origin, url)
+            if not_found:
+                # Check if origin is in the dynamic Institutions whitelist
+                if Institution.find(Q('domains', 'eq', url.netloc.lower())).count() != 0:
```

Comment:
I talked to @icereval about it a while ago and he was okay loading them on app start. I started working on it loading up inside settings but then :shipit: and I fell into something else and never looped back. It's not a big deal either way. I just remember seeing the number of calls to this query in new relic and thinking it didn't need to be there.